### PR TITLE
Feature/gene info

### PR DESCRIPTION
**Pull request by LiamSingh64**
This PR:
- Creates MutationsView which has a RadioGroup so only 1 Gene is selected at a time
- Calls the corresponding API endpoints (https://github.com/advweb-grp1/advanced-web-final-year-project/issues/48#issuecomment-1518771526) everytime the button is clicked. (Store isn't used for this reason: https://github.com/advweb-grp1/advanced-web-final-year-project/issues/48#issuecomment-1524009279)
- Resulting call is parsed for gene name, description and summary
- Parsed data is displayed in a card
- Route added to NavBar once the User has logged in

### CODE REVIEW

**Comment by LiamSingh64** [comment]
I already mentioned it in the issue and at the top of this PR but an API call is made everytime the button is clicked instead of using a Store bcos endpoints were not reliably returning data whilst 'bulk' calling and Ayman couldn't figure out why

So this is either a WIP before we change it to use a store properly OR we just keep it like this bcos it works, isn't even slow and we have other priorities for this project... and our dissertations!

**Comment by rwx-yxu** [comment]
the mutations view should be available to everyone. Not just on log in.

**Comment by rwx-yxu** [comment]
some minor styling should be fixed 
image
The card header looks odd. There is some background colour coming through on the edges.
image
Too wordy for the card. Header. Should just be the mutation name.

**Comment by AymanReh** [comment]
Final view for Gene info after styling changes:
image

File: src/views/MutationsView.vue [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  const dataReturned = computed(() => {
+    return Object.keys(searchedGene).length !== 0;
+  });
+
+  const getGeneFromCall = async () => {
+    if(geneId.value != ''){
+      await fetch('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=gene&id='+
+        geneId.value+
+        '&retmode=xml')
+        .then(response => response.text())
```
Comment (rwx-yxu) [inline src/views/MutationsView.vue:150]: Why call .then when you are in an async function that has an await on this?
Could you not do
```
try{
  const response = await fetch(`https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=gene&id=${geneId.value}&retmode=xml`);
....
}catch (error) {
  console.log(error);
}
```

Comment (AymanReh) [reply src/views/MutationsView.vue:150]: So something like this?
```
 try{
        const response = await fetch('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=gene&id='+
          geneId.value+
          '&retmode=xml');
        const data = await response.text();
        let xmlDoc = parser.parseFromString(data, 'text/xml');
        let desc = xmlDoc.getElementsByTagName('Gene-ref_desc')[0].childNodes[0].nodeValue;
        let summary = xmlDoc.getElementsByTagName('Entrezgene_summary')[0].childNodes[0].nodeValue;
        let name = xmlDoc.getElementsByTagName('Gene-ref_locus')[0].childNodes[0].nodeValue;
        searchedGene.value = { desc, summary, name };
      }
      catch(error) {
        console.log(error);
      }
```
Ive implemented it in a way that doesn't use .then and instead uses a try catch instead.
